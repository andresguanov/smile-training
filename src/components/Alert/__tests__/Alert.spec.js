import { shallowMount } from "@vue/test-utils"
import smIcon from "@components/Icon/Icon.vue"
import smAlert from "@components/Alert/Alert.vue"
let wrapper
const title = "alert title"

describe("smAlert", () => {
  //#region Basic render
  it("renders default alert", () => {
    wrapper = shallowMount(smAlert, {
      propsData: {
        title
      }
    })
    expect(wrapper.find(".is-small.title").text()).toBe(title)
    expect(wrapper.classes("bounceInRight")).toBe(true)
    expect(wrapper.classes("success")).toBe(true)
    expect(wrapper.classes("removed")).not.toBe(true)
    expect(wrapper.findComponent(smIcon).exists()).toBe(true)
    expect(wrapper.vm.timeOutCallback).not.toBeNull()
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  it("renders message", () => {
    const message = "alert message"
    wrapper = shallowMount(smAlert, {
      propsData: {
        title,
        message
      }
    })
    expect(wrapper.find(".content p.is-small:not(.title)").text()).toBe(message)
  })

  it("sets removed", () => {
    wrapper = shallowMount(smAlert, {
      propsData: {
        title,
        removed: true
      }
    })
    expect(wrapper.classes("removed")).toBe(true)
  })

  it("sets stacked", () => {
    wrapper = shallowMount(smAlert, {
      propsData: {
        title,
        stacked: true
      }
    })
    expect(wrapper.classes("stacked")).toBe(true)
  })

  it("sets type", async () => {
    wrapper = shallowMount(smAlert, {
      propsData: {
        title,
        type: "warning"
      }
    })
    expect(wrapper.classes("warning")).toBe(true)
    expect(wrapper.findComponent(smIcon).attributes("icon")).toBe("info")
    await wrapper.setProps({
      type: "error"
    })
    expect(wrapper.classes("error")).toBe(true)
    expect(wrapper.findComponent(smIcon).attributes("icon")).toBe("info")
    await wrapper.setProps({
      type: "info"
    })
    expect(wrapper.classes("info")).toBe(true)
    expect(wrapper.findComponent(smIcon).attributes("icon")).toBe("info")
    const validator = wrapper.vm.$options.props.type.validator
    expect(validator("not-valid-type")).toBeFalsy()
  })
  //#endregion

  //#region Methods
  it("closes alert", () => {
    const close = jest.fn()
    wrapper = shallowMount(smAlert, {
      propsData: {
        title,
        type: "warning"
      },
      listeners: {
        close
      }
    })
    wrapper.vm.close()
    expect(wrapper.vm.timeOutCallback).toBeNull()
    expect(wrapper.vm.show).toBe(false)
    expect(close).toHaveBeenCalledTimes(1)
  })
  //#endregion

  //#region Events
  it("calls close method on click", () => {
    const close = jest.fn()
    wrapper = shallowMount(smAlert, {
      propsData: {
        title,
        type: "warning"
      },
      listeners: {
        close
      }
    })
    wrapper.find(".close").trigger("click")
    expect(wrapper.vm.timeOutCallback).toBeNull()
    expect(wrapper.vm.show).toBe(false)
    expect(close).toHaveBeenCalledTimes(1)
  })
  //#endregion

  //#region Slots
  //#endregion
})
