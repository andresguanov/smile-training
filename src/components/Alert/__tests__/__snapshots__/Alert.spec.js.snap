// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smAlert renders default alert 1`] = `
<div class="alert bounceInRight success" style="height: 0px;">
  <div class="content">
    <div class="alert-icon">
      <icon-stub size="large" icon="success" width="" height="" color="#999999"></icon-stub>
    </div>
    <div class="text">
      <p class="is-small title">alert title</p>
      <p class="is-small"></p>
    </div>
    <div class="close">
      <icon-stub size="" icon="close" width="" height="" color="#999999"></icon-stub>
    </div>
  </div>
</div>
`;
